### PR TITLE
add sort operation support

### PR DESCRIPTION
**Pull request by asafronienkov**

### CODE REVIEW

**Comment by fppt** [comment]
Hey! Thanks for the PR. Going to take a look at it now

File: src/main/java/com/github/fppt/jedismock/operations/RO_zadd.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    Slice response() {
+        Slice key = params().get(0);
+        Slice data = base().getValue(key);
+        Set<Slice> set;
+
+        if (data != null) {
+            set = deserializeObject(data);
+        } else {
+            set = new LinkedHashSet<>();
```
Comment (fppt) [inline src/main/java/com/github/fppt/jedismock/operations/RO_zadd.java]: Bit of a nitpick but I think a standard `HashSet` might work

File: src/main/java/com/github/fppt/jedismock/operations/RO_zadd.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import jdk.nashorn.internal.runtime.regexp.joni.exception.InternalException;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.github.fppt.jedismock.Utils.deserializeObject;
+import static com.github.fppt.jedismock.Utils.serializeObject;
+
+public class RO_zadd extends AbstractRedisOperation {
```
Comment (fppt) [inline src/main/java/com/github/fppt/jedismock/operations/RO_zadd.java]: Maybe take a look at the base implementation of [add](https://github.com/fppt/jedis-mock/blob/master/src/main/java/com/github/fppt/jedismock/operations/RO_add.java)

I think you might be able to extend on that and get rid of some of the duplicate code. You can look [here](https://github.com/fppt/jedis-mock/blob/master/src/main/java/com/github/fppt/jedismock/operations/RO_sadd.java) as an example.

File: src/main/java/com/github/fppt/jedismock/operations/RO_zrange.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import com.google.common.collect.Sets;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import static com.github.fppt.jedismock.Utils.convertToInteger;
+import static com.github.fppt.jedismock.Utils.deserializeObject;
+
+public class RO_zrange extends AbstractRedisOperation {
```
Comment (fppt) [inline src/main/java/com/github/fppt/jedismock/operations/RO_zrange.java]: I am wondering if we might be able to abstract this into a common `RO_range` class. That way `RO_zrange` and `RO_lrange` might be able to get rid of some bloat. If you look at `RO_lrange` it almost does the same thing as this.

File: src/main/java/com/github/fppt/jedismock/operations/RO_zrem.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import com.github.fppt.jedismock.storage.RedisBase;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import static com.github.fppt.jedismock.Utils.deserializeObject;
+import static com.github.fppt.jedismock.Utils.serializeObject;
+
+public class RO_zrem extends AbstractRedisOperation {
```
Comment (fppt) [inline src/main/java/com/github/fppt/jedismock/operations/RO_zrem.java]: Same thing here. Maybe we can abstract into a `RO_rem` class which `RO_zrem` and `RO_lrem` can share.

File: src/main/java/com/github/fppt/jedismock/operations/RO_zadd.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            set = new LinkedHashSet<>();
+        }
+
+        for (int i = 1; i < params().size(); i++) {
+            set.add(params().get(i));
+        }
+        try {
+            base().putValue(key, serializeObject(set), -1L);
+        } catch (Exception e) {
+            throw new InternalException(e.getMessage());
```
Comment (inponomarev) [inline src/main/java/com/github/fppt/jedismock/operations/RO_zadd.java]: Why `jdk.nashorn.internal.runtime.regexp.joni.exception.InternalException` here?
Should be wrapped into RuntimeException instead

**Comment by inponomarev** [comment]
Hey @asafronienkov @fppt , it seems to me that this is not an implementation of [`ZADD`](https://redis.io/commands/zadd)/[`ZREM`](https://redis.io/commands/zrem)/[`ZRANGE`](https://redis.io/commands/zrange) at all. To store sorted sets `TreeMap<Double,String>` should be used, not `LinkedHashSet`.

**Comment by fppt** [comment]
Good point @inponomarev, PR should still be usable though

**Comment by fppt** [comment]
Closing this PR as it seems abandoned. Happy to reopen and keep working on it.

**Pull request closed without merging**